### ISSUE TITLE
Mess Code on Debugger

### ISSUE THREAD

**Issue by GeraltCN** [syl20bnr/spacemacs#8402]
```
Debugger entered--Lisp error: (void-function write)
  (write v1)
  eval((write v1) nil)
  elisp--eval-last-sexp(nil)
  #[257 "\204\303!\207	\303!\n)B\211A	=\204\211A\211@\207" [eval-expression-debug-on-error elisp--eval-last-sexp-fake-value debug-on-error elisp--eval-last-sexp] 4 2324429 "P"](nil)
  #[(&rest _it) "	!\211\207" [ad--addoit-function eval-last-sexp-arg-internal ad-return-value] 2]()
  eval-sexp-fu-flash-doit-simple(#[(&rest _it) "	!\211\207" [ad--addoit-function eval-last-sexp-arg-internal ad-return-value] 2] #[128 "\302\300\303\301\"\"\207" [esf-hl-highlight-bounds ((30 . 40) eval-sexp-fu-flash #<buffer kdtree.el>) apply append] 6 "\n\n(fn &rest ARGS2)"] #[128 "\302\300\303\301\"\"\207" [esf-hl-unhighlight-bounds ((30 . 40) #<buffer kdtree.el>) apply append] 6 "\n\n(fn &rest ARGS2)"])
  eval-sexp-fu-flash-doit(#[(&rest _it) "	!\211\207" [ad--addoit-function eval-last-sexp-arg-internal ad-return-value] 2] #[128 "\302\300\303\301\"\"\207" [esf-hl-highlight-bounds ((30 . 40) eval-sexp-fu-flash #<buffer kdtree.el>) apply append] 6 "\n\n(fn &rest ARGS2)"] #[128 "\302\300\303\301\"\"\207" [esf-hl-unhighlight-bounds ((30 . 40) #<buffer kdtree.el>) apply append] 6 "\n\n(fn &rest ARGS2)"])
  esf-flash-doit(#[(&rest _it) "	!\211\207" [ad--addoit-function eval-last-sexp-arg-internal ad-return-value] 2] #[128 "\302\300\303\301\"\"\207" [esf-hl-highlight-bounds ((30 . 40) eval-sexp-fu-flash #<buffer kdtree.el>) apply append] 6 "\n\n(fn &rest ARGS2)"] #[128 "\302\300\303\301\"\"\207" [esf-hl-unhighlight-bounds ((30 . 40) #<buffer kdtree.el>) apply append] 6 "\n\n(fn &rest ARGS2)"] #[128 "\302\300\303\301\"\"\207" [esf-flash-error-bounds ((30 . 40) #<buffer kdtree.el> eval-sexp-fu-flash-error) apply append] 6 "\n\n(fn &rest ARGS2)"])
  ad-Advice-eval-last-sexp(#[257 "\204\303!\207	\303!\n)B\211A	=\204\211A\211@\207" [eval-expression-debug-on-error elisp--eval-last-sexp-fake-value debug-on-error elisp--eval-last-sexp] 4 2324429 "P"] nil)
  apply(ad-Advice-eval-last-sexp #[257 "\204\303!\207	\303!\n)B\211A	=\204\211A\211@\207" [eval-expression-debug-on-error elisp--eval-last-sexp-fake-value debug-on-error elisp--eval-last-sexp] 4 2324429 "P"] nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

When I execute (+ 2 "asd"). The problem is the mess code on my debugger buffer. How to fix it?

**Comment by bmag** [comment]
The "mess code" is just elisp byte code, and the way it shows is an upstream Emacs issue. Please see http://stackoverflow.com/questions/26892109/how-to-fix-corrupted-text-in-backtrace-buffer-emacs and https://debbugs.gnu.org/cgi/bugreport.cgi?bug=6991.

As for the "When I execute (+ 2 "asd")" problem, please include specific repro steps and your system info (`SPC h d s`), as explained in [Debugging](https://github.com/syl20bnr/spacemacs/wiki/Debugging). I can't test the problem right now, but if you add more details then I (or someone else) can look at it later.

**Comment by GeraltCN** [comment]
#### System Info :computer:
- OS: darwin
- Emacs: 25.1.1
- Spacemacs: 0.200.7
- Spacemacs branch: master (rev. c2774bc)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(csv sql html octave helm auto-completion better-defaults emacs-lisp
     (colors :variables colors-colorize-identifiers 'all)
     git markdown org
     (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode)
     syntax-checking osx python java go)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS

**Comment by bmag** [comment]
What about repro steps?

**Comment by GeraltCN** [comment]
Just type `(+ 2 "asd")` in scratch buffer then (C-x C-e) because of the wrong code, the debugger comes out. But if I use emacs instead of spacemacs with a normal .emacs.d file. These byte codes won't appear.

**Comment by sampablokuper** [comment]
@bmag wrote:
> The "mess code" is just elisp byte code, and the way it shows is an upstream Emacs issue. Please see http://stackoverflow.com/questions/26892109/how-to-fix-corrupted-text-in-backtrace-buffer-emacs and https://debbugs.gnu.org/cgi/bugreport.cgi?bug=6991.

Bug #6991 appears to have been [fixed in June 2017](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=6991#210). Better late than never :)

However, the issue persists in Spacemacs, unfortunately.

> As for the "When I execute (+ 2 "asd")" problem, please include specific repro steps

I am not the OP, but I can reproduce this. Here are the steps.

## In Emacs 25, debugger behaves as expected

Invoke Emacs from GNU Bash with: `emacs25 -q &`.

Then `C-x b <RET>` to open the `*scratch*` buffer.

Type `(+ 2 "asd")` and then press `C-x C-e` (`eval-last-sexp`).

As desired, the debugger then opens and shows:

```
Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p "asd")
  +(2 "asd")
  eval((+ 2 "asd") nil)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

## In Spacemacs on Emacs 25, debugger gives bytecode

Invoke Spacemacs from GNU Bash with: `emacs25 &`.

Then `SPC b b <RET>` to open the `*scratch*` buffer.

Enter insert mode (if using Evil-flavored Spacemacs) and type `(+ 2 "asd")`. Then, leaving insert mode beforehand if appropriate, press `C-x C-e` (`eval-last-sexp`).

As in Emacs, the debugger then opens; but **unlike in Emacs**, it shows:

```
Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p "asd")
  +(2 "asd")
  eval((+ 2 "asd") nil)
  elisp--eval-last-sexp(nil)
  #[257 "^H\204^H^@\303^A!\207	^Z\303^A!\n)B\211A	=\204^Z^@\211A^R\211@\207" [eval-expression-debug-on-error elisp--eval-last-sexp-fake-value debug-on-error elisp--eval-last-sexp] 4 2269477 "P"](nil)
  ad-Advice-eval-last-sexp(#[257 "^H\204^H^@\303^A!\207	^Z\303^A!\n)B\211A	=\204^Z^@\211A^R\211@\207" [eval-expression-debug-on-error elisp--eval-last-sexp-fake-value debug-on-error elisp--eval-last-sexp] 4 2269477 "P"] nil)
  apply(ad-Advice-eval-last-sexp  #[257 "^H\204^H^@\303^A!\207	^Z\303^A!\n)B\211A	=\204^Z^@\211A^R\211@\207" [eval-expression-debug-on-error elisp--eval-last-sexp-fake-value debug-on-error elisp--eval-last-sexp] 4 2269477 "P"] nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

> and your system info (SPC h d s), as explained in Debugging.

```
#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.1.1
- Spacemacs: 0.200.10
- Spacemacs branch: master (rev. 4bb4cb46)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(markdown yaml helm
          (auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-complete-with-key-sequence nil auto-completion-complete-with-key-sequence-delay 0.1 auto-completion-private-snippets-directory nil)
          emacs-lisp org bibtex)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11
```

**Comment by sampablokuper** [comment]
Incidentally, this issue prevents Spacemacs from being used successfully in place of plain Emacs, for at least section 1.3 ("Generate an error message") of [*An Introduction to Programming in Emacs Lisp*](https://www.gnu.org/software/emacs/manual/eintr.html).

**Comment by Miciah** [comment]
https://debbugs.gnu.org/cgi/bugreport.cgi?bug=6991 was fixed in Emacs 26.1 but not in Emacs 25.

Using Emacs 27.0.50, evaluating `(+ 2 "asd")` gives me the following backtrace:
```
Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p "asd")
  +(2 "asd")
  eval((+ 2 "asd") nil)
  elisp--eval-last-sexp(nil)
  #f(compiled-function (eval-last-sexp-arg-internal) "Evaluate sexp before point; print value in the echo area.\nInteractively, with a non `-' prefix argument, print output into\ncurrent buffer.\n\nNormally, this function truncates long output according to the\nvalue of the variables `eval-expression-print-length' and\n`eval-expression-print-level'.  With a prefix argument of zero,\nhowever, there is no such truncation.\nInteger values are printed in several formats (decimal, octal,\nand hexadecimal).  When the prefix argument is -1 or the value\ndoesn't exceed `eval-expression-print-maximum-character', an\ninteger value is also printed as a character of that codepoint.\n\nIf `eval-expression-debug-on-error' is non-nil, which is the default,\nthis command arranges for all errors to enter the debugger." (interactive "P") #<bytecode 0x26afa5>)(nil)
  #f(compiled-function (&rest _it) #<bytecode 0x4800e41>)()
  eval-sexp-fu-flash-doit-simple(#f(compiled-function (&rest _it) #<bytecode 0x4800e41>) #f(compiled-function (&rest args2) #<bytecode 0x2f2043d>) #f(compiled-function (&rest args2) #<bytecode 0x6958ced>))
  eval-sexp-fu-flash-doit(#f(compiled-function (&rest _it) #<bytecode 0x4800e41>) #f(compiled-function (&rest args2) #<bytecode 0x2f2043d>) #f(compiled-function (&rest args2) #<bytecode 0x6958ced>))
  esf-flash-doit(#f(compiled-function (&rest _it) #<bytecode 0x4800e41>) #f(compiled-function (&rest args2) #<bytecode 0x2f2043d>) #f(compiled-function (&rest args2) #<bytecode 0x6958ced>) #f(compiled-function (&rest args2) #<bytecode 0x6958d0d>))
  ad-Advice-eval-last-sexp(#f(compiled-function (eval-last-sexp-arg-internal) "Evaluate sexp before point; print value in the echo area.\nInteractively, with a non `-' prefix argument, print output into\ncurrent buffer.\n\nNormally, this function truncates long output according to the\nvalue of the variables `eval-expression-print-length' and\n`eval-expression-print-level'.  With a prefix argument of zero,\nhowever, there is no such truncation.\nInteger values are printed in several formats (decimal, octal,\nand hexadecimal).  When the prefix argument is -1 or the value\ndoesn't exceed `eval-expression-print-maximum-character', an\ninteger value is also printed as a character of that codepoint.\n\nIf `eval-expression-debug-on-error' is non-nil, which is the default,\nthis command arranges for all errors to enter the debugger." (interactive "P") #<bytecode 0x26afa5>) nil)
  apply(ad-Advice-eval-last-sexp #f(compiled-function (eval-last-sexp-arg-internal) "Evaluate sexp before point; print value in the echo area.\nInteractively, with a non `-' prefix argument, print output into\ncurrent buffer.\n\nNormally, this function truncates long output according to the\nvalue of the variables `eval-expression-print-length' and\n`eval-expression-print-level'.  With a prefix argument of zero,\nhowever, there is no such truncation.\nInteger values are printed in several formats (decimal, octal,\nand hexadecimal).  When the prefix argument is -1 or the value\ndoesn't exceed `eval-expression-print-maximum-character', an\ninteger value is also printed as a character of that codepoint.\n\nIf `eval-expression-debug-on-error' is non-nil, which is the default,\nthis command arranges for all errors to enter the debugger." (interactive "P") #<bytecode 0x26afa5>) nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```
I think this issue can be labeled as "Fixed upstream" and "Emacs 25.1".
